src/System.Runtime.Serialization.Xml/tests/CanonicalizationTestHelper.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Text;
+using System.Xml;
+
+namespace System.Runtime.Serialization.Xml.Canonicalization.Tests
+{
+    public class ReaderWriterFactory
+    {
+        public enum ReaderWriterType
+        {
+            Binary,
+            Text,
+            MTOM,
+            WebData,
+            WrappedWebData
+        };
+
+        public static ReaderWriterType Binary = ReaderWriterType.Binary;
+        public static ReaderWriterType Text = ReaderWriterType.Text;
+        public static ReaderWriterType MTOM = ReaderWriterType.MTOM;
+        public static ReaderWriterType WebData = ReaderWriterType.WebData;
+        public static ReaderWriterType WrappedWebData = ReaderWriterType.WrappedWebData;
+
+        public static XmlReader CreateXmlReader(ReaderWriterType rwType, byte[] buffer, Encoding encoding, XmlDictionaryReaderQuotas quotas, IXmlDictionary dictionary, OnXmlDictionaryReaderClose onClose)
+        {
+            XmlReader result = null;
+            switch (rwType)
+            {
+                case ReaderWriterType.Binary:
+                    result = XmlDictionaryReader.CreateBinaryReader(buffer, 0, buffer.Length, dictionary, quotas, null, onClose);
+                    break;
+                case ReaderWriterType.Text:
+                    result = XmlDictionaryReader.CreateTextReader(buffer, 0, buffer.Length, encoding, quotas, onClose);
+                    break;
+                case ReaderWriterType.WebData:
+                    if (quotas != XmlDictionaryReaderQuotas.Max)
```

Comment:
It doesn't relate to test case logic itself, so I think we can keep it as it is...